classes/class-base.php

```diff
@@ -1551,7 +1551,7 @@ public function parse_url( $url, $component = -1 ) {
 			$url = ( \is_ssl() ? 'https://' : 'http://' ) . $url;
 		}
 		// Because encoded ampersands in the filename break things.
-		$url = \str_replace( '&#038;', '&', $url );
+		$url = \str_replace( '&#038;', '&', \html_entity_decode( $url ) );
```

Comment:
This str_replace should be dropped in favor of the html entity decode which already does that as there is a small risk of double decoding

This is not the case for the other ones because they pass through `esc_url`

Reply:
Ah, that's true, the only place we still need the str_replace logic is after esc_url().